ops/pebble.py

```diff
@@ -627,19 +627,26 @@ def wait_change(
         raise TimeoutError(
             'timed out waiting for change {} ({} seconds)'.format(change_id, timeout))
 
-    def add_layer(self, layer: Union[str, dict, Layer]):
-        """Dynamically add a layer to the Pebble setup."""
+    def merge_layer(self, layer: Union[str, dict, Layer]):
+        """Dynamically merge layer onto the Pebble setup layers.
+
+        Pebble merges the layer with the existing dynamic layer according to
+        its flattening/override rules, or adds a new dynamic if there are no
+        dynamic layers.
+        """
         if isinstance(layer, str):
             layer_yaml = layer
         elif isinstance(layer, dict):
             layer_yaml = Layer(layer).to_yaml()
         else:
```

Comment:
This should check if its an instance of Layer

Reply:
Done and added test for this - thanks.